### PR TITLE
Adding Led control package

### PR DESCRIPTION
**Pull request by FernandezR**
Migrating LED control package from bwi_common to segbot.

Package controls led strips on segbots and should not conflict with other packages in segbot.

@jack-oquin Does this look good for merging in to master?

### CODE REVIEW

**Comment by FernandezR** [comment]
@jack-oquin Do you have any idea why this would fail here and not in common? I though I had fixed the problem when it happened in common but it's occurring again.

**Comment by jack-oquin** [comment]
I have not looked at it yet in detail, but maybe it depends on something in bwi_common that is not yet released. 

Where is ``led_clear.h`` defined? If it's a service, maybe you forgot the necessary ``add_dependencies()``?

I plan to release an up-to-date bwi_common soon, maybe next week.

**Comment by FernandezR** [comment]
It is a service file and the add depends is there, so I don't know why it happens in segbot now and not in bwi common.

**Comment by jack-oquin** [comment]
As mentioned in utexas-bwi/bwi_common#63, all the interface definitions probably ought to go in ``bwi_msgs``, but that will prevent this from being merged until after a new ``bwi_common`` release. 

That does not explain the current Travis failure, however. I'll see if I can figure that out.

File: segbot_led/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(segbot_led)
+find_package(catkin REQUIRED COMPONENTS
+  roscpp
+  serial
+  std_msgs
+  actionlib
+  actionlib_msgs
+  message_generation
```
Comment (jack-oquin) [inline segbot_led/CMakeLists.txt]: Alphabetical order is better.

File: segbot_led/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+add_definitions(-std=c++11)
+
+add_service_files(FILES 
+				  led_clear.srv
+				  test_strip.srv
+				  set_first_five.srv
+				  set_every_fifth.srv
+          set_camera_on.srv
+          set_camera_off.srv)
+
```
Comment (jack-oquin) [inline segbot_led/CMakeLists.txt]: It would be nice to indent these files consistently and alphabetize the names.

**Comment by jack-oquin** [comment]
I think this line:
```
add_dependencies(led_control_server ${catkin_EXPORTED_TARGETS})
```
would also need to specify ``${${PROJECT_NAME}_EXPORTED_TARGETS}``, as [described in the catkin howto](http://docs.ros.org/jade/api/catkin/html/howto/format2/building_msgs.html).

That's in order to resolve the dependency on interfaces defined within this same package. Of course, moving the services and actions to ``bwi_msgs`` will make that change unnecessary.

**Comment by jack-oquin** [comment]
I am waiting to merge this until after the pending bwi_common release, so the Travis tests have a reasonable chance of succeeding.

**Comment by jack-oquin** [comment]
Integration still failing:
```
[100%] Building CXX object CMakeFiles/led_control_server.dir/src/led_control_server.cpp.o
/home/travis/catkin_ws/src/segbot/segbot_led/src/led_control_server.cpp:20:31: fatal error: bwi_msgs/LEDClear.h: No such file or directory
 #include "bwi_msgs/LEDClear.h"
                               ^
compilation terminated.
```

Is this another ``add_dependencies()`` problem?

**Comment by FernandezR** [comment]
I'll take a look at it, I thought they were all addressed as it compiles fine on the machines. Could Travis not be pulling the most recent bwi common release?

**Comment by jack-oquin** [comment]
Maybe not. I just released it this morning.

**Comment by jack-oquin** [comment]
The new bwi_common was already in the ``ros-shadow-fixed`` repository, but Travis was configured to use ``ros``, instead. I changed that in commit ca1ce17.

**Comment by jack-oquin** [comment]
Looks good now.

**Pull request merged**